docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,428 @@
+| [`server.address`](../general/attributes.md) | string | Logical server hostname, matches server FQDN if available, and IP or socket address if FQDN is not known. | `example.com` | Recommended |
+| [`server.port`](../general/attributes.md) | int | Logical server port number | `80`; `8080`; `443` | Recommended |
+| [`server.socket.address`](../general/attributes.md) | string | Physical server IP address or Unix socket address. If set from the client, should simply use the socket's peer address, and not attempt to find any actual server IP (i.e., if set from client, this may represent some proxy server instead of the logical server). | `10.5.3.2` | Recommended: If different than `server.address`. |
+| [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `https`; `ftp`; `telnet` | Recommended |
+
+**[1]:** **TODO: there is a open discussion/proposal in otel for http error code - https://github.com/open-telemetry/semantic-conventions/pull/205**
+
+**[2]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+`dotnet.http.error.reason` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `timeout` | Operation timed out |
+| `cancellation` | Operation was cancelled |
+| `unknown_error` | Unknown error |
+| `name_resolution_error` | NameResolutionError |
+| `connection_error` | ConnectionError |
+| `secure_connection_error` | SecureConnectionError |
+| `http_protocol_error` | HttpProtocolError |
+| `extended_connection_not_supported` | ExtendedConnectNotSupported |
+<!-- endsemconv -->
+
+### Metric: `http.client.duration`
+
+**TODO: should we add error status to http semconv ?**  - https://github.com/open-telemetry/opentelemetry-specification/issues/3243
+
+<!-- semconv metric.dotnet.http.client.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.duration` | Histogram | `s` | Measures the duration of outbound HTTP requests. |
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.client.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `dotnet.http.error.reason` | string | Request error such as timeout, cancellation or one of [HTTP Request error](https://github.com/dotnet/runtime/blob/c430570a01c103bc7f117be573f37d8ce8a129b8/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs) [1] | `timeout`; `http_protocol_error` | Recommended |
```

Comment:
Can you elaborate why is this `dotnet.http.error.reaason` and not `http.request.error.reason`? Is it because we don't expect https://github.com/open-telemetry/semantic-conventions/pull/205 to land in a timely manner?

Reply:
yeah, my understanding is that we need to finalize names here by Friday (7/28) and I'd expect otel change to take a bit longer - I'll address your comments on it and mark it ready for the review tomorrow.

I can call it `http.error.reason` both here and in the otel proposal. `dotnet` in front of it protects us from any conflicts with anything that will be defined/changed before stability in otel.

Reply:
If can at least agree on a stable property name, future changes will be less breaking (only impact the potential values, not the name), do we think it's not realistic? If we named it `dotnet.http.error.reaason` we would need to rename it in the future anyways, so we would be in the same situation as if we guessed the name wrong. @noahfalk thoughts?

Reply:
Two thoughts:
1. In the other thread I suggested droppping the 'client' from the name but I think that was a mistake. The implementations and common errors that come from clients and servers are probably different enough on average that even though they may have some common values in an enumeration its probably not effective to combine them. Also handling of http server inbound requests often includes sending http client outbound requests. If the reason an http server request failed is because the nested outbound request had an error using a shared set of errors may create some confusion how that error is supposed to be represented.
2. I have a fair bit of skepticism of around error abstractions that attempt to be implementation neutral, more so if implementations are unlikely to adopt the abstraction as their internal error represention and expose it consistently. I'd expect users looking at the telemetry will generally want to map it back to implementation specific errors in order to make better sense of it or to leverage documentation and guidance that already exists for their technology. The implementation neutral errors are likely to be lower granularity and the mapping back may be lossy or non-obvious.
All this to say I'd be just fine leaving this as `dotnet.http.client.error`. If OTel defines a impl-neutral error in the future we could add it once it was stable, but I don't know that we would necesarily remove the dotnet one.

Reply:
the idea behind `http.error.reason` that every instrumentation can put anything it wants there. 
E.g. .NET HTTP client would put `http_protocol_error` and ASP.NET Core would put `client_closed_connection`. The point is that we/users can build dashboards or write queries and groups by error reason.

users can  write:
-  `rate(http.client.duration{service.name="foo", http.status.code="0"}) by (http.error.reason)` and would get all client errors 
or 
- `rate(http.server.duration{service.name="foo", http.status.code="0"}) by (http.error.reason)` and get all server errors.

I.e. there is no need to agree on error reasons between .NET and Java HTTP clients - they are coming from different metrics.

Still, it would be desirable if .NET client and server reported "timeout", "cancellation" and other (if any) common errors consistently.

If we do `dotnet.http.client.error`, we'd also need `dotnet.http.server.error` and would need to align them. I don't see a benefit in this if we can just document applicable error reasons for each instrumentation individually.

Reply:
OK, I misunderstood the premise it sounds like. If implementations are really free to put whatever string they want in http.error.reason and we aren't trying to align around common enumeration values I'm fine using `http.error.reason` as the attribute name.